### PR TITLE
Fix bug 1714 null serialisation by default

### PR DESCRIPTION
**Pull request by biddster**
Attempt to fix attribute mapping so it respects the assignsDefaultValueForMissingAttributes value.

New test method included and the entire suite still passes.

### CODE REVIEW

File: Code/ObjectMapping/RKObjectMapping.m [hunk 1]
```diff
@@ -69,8 +69,8 @@ - (RKObjectMapping *)invertMapping:(RKObjectMapping *)mapping withPredicate:(BOO
     inverseMapping = [RKObjectMapping requestMapping];
     (self.invertedMappings)[dictionaryKey] = inverseMapping;
     [inverseMapping copyPropertiesFromMapping:mapping];
-    // We want to serialize `nil` values
-    inverseMapping.assignsDefaultValueForMissingAttributes = YES;
+    // We potentially want to serialize `nil` values, adopt the strategy from the mapping we're inverting.
```
Comment (segiddins) [inline Code/ObjectMapping/RKObjectMapping.m]: why this change, it seems unrelated?

Comment (biddster) [inline Code/ObjectMapping/RKObjectMapping.m]: Arguably. If I create a mapping with assignsDefaultValueForMissingAttributes = NO and I ask for the inverse, I wouldn't expect assignsDefaultValueForMissingAttributes to suddenly be YES. I'd expect the mirror of the mappings and settings.

Comment (segiddins) [inline Code/ObjectMapping/RKObjectMapping.m]: Can you check the blame on that line & the comment and make sure it isn't that way for a reason?

Comment (segiddins) [inline Code/ObjectMapping/RKObjectMapping.m]: That's intentional, please revert this part of your PR and I can merge. https://github.com/RestKit/RestKit/commit/1177f8075cce0bf1d62f849b8408ad2ebe23bb17

Comment (biddster) [inline Code/ObjectMapping/RKObjectMapping.m]: Ok, Reverted and pushed.

What's the intent? Do we want to update that comment with a little more information?

**Comment by biddster** [comment]
Expand the outdated diff above. That behaviour was intentional (assignsDefaultValueForMissingAttributes=YES) and I was asked to revert that part of the PR by @segiddins before it could be merged.

**Comment by segiddins** [comment]
Thanks so much @biddster !

**Comment by biddster** [comment]
Pleasure @segiddins, thanks for the merge :-)
